evalml/pipelines/components/transformers/preprocessing/decomposer.py

```diff
@@ -220,6 +221,19 @@ def set_seasonal_period(self, X: pd.DataFrame, y: pd.Series):
         self.seasonal_period = self.determine_periodicity(X, y)
         self.parameters["seasonal_period"] = self.seasonal_period
 
+    def _choose_proper_index(self, y):
+        """Function that provides support for targets with integer and datetime indices."""
+        # TODO: Need to update this after we upgrade to Pandas 1.5+ and Int64Index is deprecated.
+        if isinstance(y.index, (Int64Index, pd.RangeIndex)):
+            index = self.original_index
```

Comment:
Does the PolynomialDecomposer have `self.original_index` to return?

Reply:
Will fix this - have a separate branch/PR that is bringing PD up to complete parity.